packages/filesystem/src/node/filesystem-watcher-client.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as path from "path";
+import { injectable, inject } from "inversify";
+import { JsonRpcProxyFactory, ILogger, ConnectionErrorHandler, DisposableCollection, Disposable } from "@theia/core";
+import { IPCConnectionProvider } from "@theia/core/lib/node/messaging";
+import { FileSystemWatcherServer, WatchOptions, FileSystemWatcherClient, ReconnectingFileSystemWatcherServer } from "../common/filesystem-watcher-protocol";
+
+export const CHOKIDAR_WATCHER = 'chokidar-watcher';
+
+@injectable()
+export class FileSystemWatcherServerClient implements FileSystemWatcherServer {
```

Comment:
File should be named `filesystem-watcher-server-client.ts` ?

Reply:
Even it it does look weird hehe, I wonder if WatcherServerProxy would be more appropriate ? or WatcherServerIPCProxy

Reply:
I am also struggling with naming it, proxy is also overused in our context :(